Fix clipping of wide vega tooltip tables

peterschretlen opened the pull request:
## Summary

Fixes #33436.  An EUI tooltip has a maximum width of 256px (defined in EUI).  
But vega tooltip tables (used when rendering key-value pairs of an object) can be up to 400px wide (160px for the key and 240px for the value). 

When table rows became wide, they appeared truncated:
![image](https://user-images.githubusercontent.com/7903188/54574996-d78ade80-49c8-11e9-8c9a-453b2a6a6ae5.png)

This change makes the max width of a vega tooltip the same as the table, giving a result like this when both the key and value exceed the max width:
![image](https://user-images.githubusercontent.com/7903188/54576024-ee333480-49cc-11e9-8892-d8f69cc96ece.png)

There is 12px on either side to account for padding
![image](https://user-images.githubusercontent.com/7903188/54576078-26d30e00-49cd-11e9-8086-53e652cb4002.png)

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
  - [x] IE11 / Windows 8.1
  - [x] Chrome 72.0.3626.121 / macOS
  - [x] Firefox 65.0.2 / macOS
- [ ] ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- [ ] ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [ ] ~~[Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
- [ ] ~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [ ] ~~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
- [ ] ~~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5935/)

cchaos commented:
> vega visualizations don't seem to load at all in IE11 (due to URL length issues)

You can test IE11 by going into Management > Advanced Settings > [Search for Session] and turn on Store Session in Local Storage

peterschretlen commented:
Thanks @cchaos, I changed the local storage setting and was able to verify the change in IE11.

cchaos commented on src/legacy/core_plugins/vega/public/_vega_vis.scss:
```diff
@@ -118,12 +122,12 @@
   td.key {
     color: $euiColorMediumShade;
-    max-width: $euiSize * 10;
+    max-width: $vegaTooltipMaxWidth * 0.4;
```
These calculations are interesting, however since you're now calculating a percentage, I think it's best to actually add the column width directly to the table column as a percentage. Like `<td width="40%">` That way if we need to readjust the sizing of the tooltip the table should auto-adjust.

peterschretlen replied:
I've tried it with width percentages on the `<td>`  elements but it required using a fixed table layout and setting the table width to 100%, like this:
```
.vgaVis__tooltip table {
    table-layout: fixed;
    width: 100%;
}
```
The width is needed in order to overflow the cells.  The problem is the table ends up being full width even if it doesn't need to be? 

With 100% width it looks fine for a wide tooltip:
![image](https://user-images.githubusercontent.com/7903188/54652252-9c51e380-4a8c-11e9-9e6c-2ef0f615ec34.png)

But a small tooltip ends up with a lot of whitespace:
![image](https://user-images.githubusercontent.com/7903188/54652383-16826800-4a8d-11e9-8ec9-7769ba016e09.png)

cchaos replied:
Ah ok, thanks for investigating that.

cchaos commented on src/legacy/core_plugins/vega/public/_vega_vis.scss:
```diff
@@ -106,6 +106,10 @@
 .vgaVis__tooltip {
+
+  $vegaTooltipMaxWidth: $euiSize * 25;
+  max-width: $vegaTooltipMaxWidth + $euiSizeL;
```
This is not going to fit on a mobile device of only `320px` wide. We need to be able to set a max-width of `100%` when below the desktop max-width needs.

elasticmachine commented:
Pinging @elastic/kibana-app

peterschretlen commented:
I've revised the change - left the cell widths as they were, and changed the width of the div to be max 100%.  To support mobile I've added a media query so the max width is 256px.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6096/)

cchaos commented on src/legacy/core_plugins/vega/public/_vega_vis.scss:
```diff
@@ -106,6 +106,9 @@
 // Adapted from https://github.com/vega/vega-tooltip
 
 .vgaVis__tooltip {
+
```
Remove extra space

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6190/)
